src/sync/push.rs

```diff
@@ -53,7 +53,7 @@ pub trait Pusher {
         push_url: &str,
         push_auth: &str,
         request_id: &str,
-    ) -> Result<PushResponse, PushError>;
+    ) -> Result<(Option<PushResponse>, HttpRequestInfo), PushError>;
```

Comment:
Would you mind adding a comment here to the trait or fetchpusher push implementation that says under what circumstances it errors? I think it errors only if there is a problem sending the request or parsing the response on a 200. Any non-200 response is Ok with an empty pushresponse.